### ISSUE TITLE
fastrtpsgen does not allow setting -t parameter

### ISSUE THREAD

**Issue by TorMFinn** [eProsima/Fast-RTPS#178]
It seems like the -t parameter is not implemented, but is mentioned with fastrtps -help
We need it because many users compile at the same time and permission issues appear with /tmp

The Following Code fixes the problem:

```
} else if (arg.equals("-t")) {
    if (count < args.length) {
        m_tempDir = Utils.addFileSeparator(args[count++]);
    } else {
        throw new BadArgumentException("No temporary directory specified after -t argument");
}
```

Added after "-d" argument handling on line 198 in fastrtpsgen.java

**Comment by TorMFinn** [comment]
Any thoughts on this? Could you add this to the repo so we don't have to patch the code ourselves?

**Comment by TorMFinn** [comment]
Here is the patch

```
diff --git a/fastrtpsgen/src/com/eprosima/fastrtps/fastrtpsgen.java b/fastrtpsgen/src/com/eprosima/fastrtps/fastrtpsgen.java
index e0476731..5ec43b70 100644
--- a/fastrtpsgen/src/com/eprosima/fastrtps/fastrtpsgen.java
+++ b/fastrtpsgen/src/com/eprosima/fastrtps/fastrtpsgen.java
@@ -195,6 +195,12 @@ public class fastrtpsgen {
                                } else {
                                        throw new BadArgumentException("No URL specified after -d argument");
                                }
+                       } else if (arg.equals("-t")) {
+                              if (count < args.length) {
+                                       m_tempDir = Utils.addFileSeparator(args[count++]);
+                              } else {
+                                       throw new BadArgumentException("No temporary directory specified after -t argument");
+                              }
                        } else if (arg.equals("-version")) {
                                showVersion();
                                System.exit(0);

```

**Comment by raykraft** [comment]
I just encountered this defect as well. I was trying to use this feature to circumvent an issue with parallel builds of structs with the same name (in different modules).

Any chance this -t capability as proposed in the patch by @TorMFinn can be added?

**Comment by MiguelCompany** [comment]
We have some issues with requests related with fastrtpsgen that we would like to address at the same time. This one will be definitely included, but we don't have an estimated date for it. We will keep you informed here.

**Comment by MiguelCompany** [comment]
This was included on commit 4d5f197385b716fe6249003f141174b95c01e7ed
